contribute/setup-mergetool.md

```diff
@@ -90,6 +90,78 @@ sudo apt install git-gui
 
 There are many tools available for this purpose, but `git gui` is one of the simplest and most effective.
 
+### Additional Git tuning
+
+#### Git configuration
+
+Increase default renamed file detection limit with mergetool and difftool
+```shell
+git config --global merge.renameLimit 2048
+git config --global diff.renameLimit 2048
+```
+
+Enable [rerere](https://git-scm.com/book/en/v2/Git-Tools-Rerere), "reuse recorded resolution" for Git.
+```shell
+git config --global rerere.enabled true
```

Comment:
<img width="736" alt="Screenshot 2024-03-05 at 6 19 53 PM" src="https://github.com/apache/pulsar-site/assets/9302460/66fee9a8-2aa7-48b1-b45d-7e5eb9fa20f1">

Reply:
> Would it make sense to commit the local git config to `<pulsar_repo_root>/.gitconfig` instead of configuring it globally for each user?

I think it's better that everyone handles their git config since usually there are environment specific settings and it would be a rabbit hole to start holding everyone by hand in every single problem. My goal is to give some advice and tips for the ones that are currently using unproductive ways. Some might have been doing this for 10 years or more without thinking of better ways. 

There are a lot of different tastes for development tooling and it's really hard to say that everyone must be using the same tools or git configuration.